File: 0747-min-cost-climbing-stairs/0747-min-cost-climbing-stairs.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        if (cost.size() == 2) {
            return min(cost[0], cost[1]);
        }

        vector<int> res(cost.size(), 0);
        res[0] = cost[0];
        res[1] = cost[1];
        
        for (int i = 2; i < res.size(); i++) {
            res[i] = cost[i] + min(res[i - 1], res[i - 2]);
        }

        return min(res[res.size() - 1], res[res.size() - 2]);
    }
};

File: 0747-min-cost-climbing-stairs/0747-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        if (cost.length == 2) {
            return Math.min(cost[0], cost[1]);
        }

        int[] total = new int[cost.length];
        total[0] = cost[0];
        total[1] = cost[1];

        for (int i = 2; i < total.length; i++) {
            total[i] = cost[i] + Math.min(total[i - 1], total[i - 2]);
        }

        return Math.min(total[total.length - 1], total[total.length - 2]);
    }
}